terraform-docker/apply/main.yaml

```diff
@@ -52,6 +61,55 @@ stages:
                         artifact: "${{ env.name }}.enc"
                         path: $(Build.Repository.LocalPath)/${{ parameters.terraformFolder }}/.terraform/plans
 
+                    - task: AmazonWebServices.aws-vsts-tools.AWSShellScript.AWSShellScript@1
+                      displayName: "Get AWS credentials"
+                      inputs:
+                        workingDirectory: "."
+                        awsCredentials: ${{ format(parameters.awsServiceConnectionTemplate, env.name) }}
+                        regionName: ${{ parameters.awsRegion }}
+                        scriptType: inline
+                        inlineScript: |
+                          set -e
+                          AWS_CONFIG_FILE=${HOME}/.aws/config
+                          AWS_SHARED_CREDENTIALS_FILE=${HOME}/.aws/credentials
+                          TMPFILE=${HOME}/.aws/tmp.json
+                          mkdir $(dirname $TMPFILE)
+                          aws sts get-federation-token --name xkf-$ENV --policy-arns arn="$AWS_ARN" --region $AWS_REGION > $TMPFILE
+
+                          KEY_ID=$(cat $TMPFILE |jq -r .Credentials.AccessKeyId)
+                          KEY_SECRET=$(cat $TMPFILE |jq -r .Credentials.SecretAccessKey)
+                          SESSION_TOKEN=$(cat $TMPFILE |jq -r .Credentials.SessionToken)
+
+                          # Create the credentials file from AWS session token
+                          cat <<EOF > $AWS_SHARED_CREDENTIALS_FILE
+                          [xkf-$ENV]
+                          aws_access_key_id = $KEY_ID
+                          aws_secret_access_key = $KEY_SECRET
+                          aws_session_token = $SESSION_TOKEN
+
+                          EOF
+
+                          # Create the config file
+
+                          cat <<EOF > $AWS_CONFIG_FILE
+                          [default]
+                          region = $AWS_REGION
+                          output = json
+
+                          [profile xkf-$ENV]
+                              source_profile = xkf-$ENV
+                              role_arn = $AWS_ARN
+
+                          EOF
```

Comment:
If I remember correctly, these settings can be set using environment variables. Would it be easier to set all settings in a .env file that is passed to docker?

Reply:
Yes it's possible, the reason why I went with this solution is that it's the solution that we use in the Makefile in the day to day activities and I would prefer keeping the pipeline and those the same.

Doing it this way we don't have to add even more environment variables to be able to use the Makefile, the users can configure there config files how ever they want and they don't have to think about what environment variables we allow.
There are kind of many possible environment variabels to configure AWS cli.
This way also makes us future proof.
We also use config files with Azure and to me it seems like it works fine.

So I think we should stick with this solution. Sounds resonable?